ops/travis/unit-testing/node-linting

```diff
@@ -1,43 +1,56 @@
 #!/bin/bash
 
-FOLDER_CHECKS=".js .eslintrc"
+# die on non 0 exit code
+set -e
+
+FOLDER_CHECKS=(js eslintrc json)
 
 before_install() {
-    if [ -n "$(bash ops/travis/helpers/test_for_changes $FOLDER_CHECKS)" ];
+    if bash ops/travis/helpers/test_for_changes "${FOLDER_CHECKS[@]}";
     then
         bash ops/travis/helpers/show_system_versions
     fi
 }
 
 install() {
-    if [ -n "$(bash ops/travis/helpers/test_for_changes $FOLDER_CHECKS)" ];
+    if bash ops/travis/helpers/test_for_changes "${FOLDER_CHECKS[@]}";
     then
         npm install -g npm@latest
-        npm install --save-dev eslint
-        npm install --save-dev babel-eslint
-        npm install --save-dev eslint-config-qx
-        npm install --save-dev eslint-plugin-qx-rules
+        npm install
+        cd services/web/client
+        npm install
+        node_modules/.bin/qx contrib update
+        node_modules/.bin/qx contrib install
     fi
 }
 
 before_script() {
-    if [ -n "$(bash ops/travis/helpers/test_for_changes $FOLDER_CHECKS)" ];
+    if bash ops/travis/helpers/test_for_changes "${FOLDER_CHECKS[@]}";
     then
         eslint --version
     fi
 }
 
 script() {
-    if [ -n "$(bash ops/travis/helpers/test_for_changes $FOLDER_CHECKS)" ];
+    if bash ops/travis/helpers/test_for_changes "${FOLDER_CHECKS[@]}";
     then
         npm run linter
+        cd services/web/client
+        node_modules/.bin/qx serve --machine-readable --target=build  --listen-port=8080 &
+        SERVER_PID=$!
+        while ! nc -z localhost 8080; do
+          sleep 1 # wait for 10 second before check again
+        done
+        node build-output/resource/qxl/testtapper/run.js --diag --verbose http://localhost:8080/build/testtapper/
```

Comment:
For clarity in the test report, shouldn't we rename this stage to something like `front-end` or `client` instead of node-linting?

![image](https://user-images.githubusercontent.com/32402063/53491612-1f3bdb80-3a97-11e9-8880-4d2833725d4b.png)

Reply:
I agree with @pcrespov the name shall be changed.

Reply:
will do :)